### PR TITLE
 Turn off weekly inactive SMS messages based on warehouse lag

### PR DESCRIPTION
**Pull request by sravfeyn**
@calellowitz @snopoke 

This is the PR for the ICDS Inactive SMS P1 ticket https://dimagi-dev.atlassian.net/browse/ICDS-741

Cal, can you validate the way I did lag calculation please. This is how my thinking was. 

Any successful  run on airflow will create a `Batch` object that records `completed_on` and the date range for the data considered for that batch via the attributes `start_datetime` and `end_datetime`. The form_staging table uses this `end_datetime` (via `corehq.warehouse.dbaccessors.get_forms_by_last_modified`) to fetch forms of a given range. So the latest form data available in the warehouse (for the complete app_status_fact) is the latest Batch's `end_datetime`.

I did some stats on existing batch runs below, and the batch duration is mostly within 24 hours (except the one that's still running now). This will be the minimum lag seen by the SMS trigger task.
```
In [14]: bl = list(Batch.objects.filter(dag_slug='app_status_batch', completed_on__isnull=False).order_by('-completed
    ...: _on'))

In [15]: [(b.completed_on-b.end_datetime).total_seconds()/(60*60) for b in bl[0:20]]
Out[15]: # In hours
[11.661836730833333,
 12.504099592777777,
 22.453406085555557,
 7.658533924166667,
 19.97456774611111,
 17.549148881111112,
 7.290091748055556,
 21.39826220111111,
 23.25774011888889,
 24.297149540833335,
 9.092858160555556,
 21.462363166666666,
 5.543469443055556,
 22.549744259166665,
 10.066691982222222,
 7.8351759238888885,
 3.4097954291666666,
 8.06512696361111,
 5.656455826111111,
 5.636946606944445]
```

### CODE REVIEW

**Comment by sravfeyn** [comment]
@ctsims  Since we can't guarantee zero lag for the warehouse data, what do you think is the acceptable lag for triggering weekly SMSes? (A lag of 24 hours means, we are sending SMS based on 24 hour old data) 

The default acceptable lag I have set now is 24 hours (random, but a bit based on last 20 batch runs). This can be overridden in localsettings without having to deploy (so we don't have to block this PR on that).

File: custom/icds/messaging/indicators.py
Comment (sravfeyn) [inline custom/icds/messaging/indicators.py]: It will be nice to log when this skips the message, but the way it's setup, we would log a message for each user.

Comment (sravfeyn) [reply custom/icds/messaging/indicators.py]: There might be some way to poll this info and send to datadog, I can do that in a separate PR latter.

**Review by calellowitz — APPROVED** [review]
Looks fine, although offlined with Sravan and would advocate for a 48 hour window

**Comment by ctsims** [comment]
I don't think I have enough context / perspective on the 24/48 question, unfortunately, I think Cal likely has the most insight on it, so i'll trust his call.

**Comment by sravfeyn** [comment]
GS should be taking the call on that 'Should/shouldn't send SMS if we don't have X hours fresh data'. Or they should at least be informed.

**Pull request merged**
